### PR TITLE
[CodeQuality] Add MoveAttributeUnderCommentRector

### LINKED ISSUE #7225: [CodeQuality] Add rule to move attributes under comment
The attributes are part of code, while docblock is comment above them. The other order will crash.

This was addressed in PHP FIG Per coding style:
https://github.com/php-fig/per-coding-style/issues/8

And php-parser: https://github.com/nikic/PHP-Parser/issues/762

<br>

Looking at code sample, maybe we could deal with this mess with Rector rule :) 

```diff
-#[Attribute]
 /**
  * @property-read $name 
  */
+#[Attribute]
 class SomeClass
 {
 }
```

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/7225

### CODE REVIEW

**Comment by samsonasik** [comment]
The `$node->attrGroups` seems always detected, and set null of `ORIGINAL_NODE` will make comment removed.

**Comment by samsonasik** [comment]
The `AttributeKey::PHP_DOC_INFO` got printed as  empty:

```
'/**\n
  *\n
  */'
```

on the following:

```
#[Attribute]
/**
 * @property-read $name
 */
```

**Comment by samsonasik** [comment]
@TomasVotruba it seems not possible, even check printing, it got removed comment:

```
        $print = $this->nodePrinter->print($node);
        echo $print;
```

got:

```
[\Rector\Tests\CodeQuality\Rector\ClassLike\MoveAttributeUnderCommentRector\Fixture\Attribute]
class SomeClass
{
```

which comment removed.

File: rules/CodeQuality/Rector/ClassLike/MoveAttributeUnderCommentRector.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        $phpDocNode = $phpDocInfo->getPhpDocNode();
+
+        if ($phpDocNode->children !== []) {
+            return null;
+        }
+
+        $comments = $node->getAttribute(AttributeKey::COMMENTS);
+        dump($comments); // comment got null
+
+        $node->attrGroups = []; // removin attrGroups remove the comment after it
```
Comment (samsonasik) [inline rules/CodeQuality/Rector/ClassLike/MoveAttributeUnderCommentRector.php:75]: @TomasVotruba when Attribute is under comment, the comments attribute got null, and removing attrGroups remove its attrGroups and comment below it.

**Comment by TomasVotruba** [comment]
@samsonasik What is missing here to get it to merge?

**Comment by samsonasik** [comment]
Attribute seems belong to next Stmt which not a `Nop` Stmt, which make it invalid empty comment when pull phpdocinfo data.

Interestingly, when Attribute removed, next `Nop` got removed.

**Comment by TomasVotruba** [comment]
Interesting, I'm just checking this rule and it seems the php-parser is moving comments somewhere outside.
It also seems the `Class_` picks up the attributes even if they're separated by a comment :+1:  That's what this rule should have fixed, so if it's already fixed in php-parser, there is no need for this one.

**Pull request closed without merging**